### PR TITLE
Port repr128-dwarf run-make test to rmake

### PR DESCRIPTION
**Pull request by beetrees**
This PR ports the repr128-dwarf run-make test to rmake, using the `gimli` crate instead of the `llvm-dwarfdump` command.

Note that this PR changes `rmake.rs` files to be compiled with the 2021 edition (previously no edition was passed to `rustc`, meaning they were compiled with the 2015 edition). This means that `panic!("{variable}")` will now work as expected in `rmake.rs` files (there's already a usage in the [wasm-symbols-not-exported test](https://github.com/rust-lang/rust/blob/aca749eefceaed0cda19a7ec5e472fce9387bc00/tests/run-make/wasm-symbols-not-exported/rmake.rs#L34) that this will fix).

Tracking issue: #121876

### CODE REVIEW

File: tests/run-make/repr128-dwarf/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,67 @@
+//@ ignore-windows
+// This test should be replaced with one in tests/debuginfo once GDB or LLDB support 128-bit enums.
```
Comment (Nilstrieb) [inline tests/run-make/repr128-dwarf/rmake.rs:2]: has this already happened maybe?

File: src/tools/compiletest/src/runtest.rs [hunk 2]
```diff
@@ -3824,6 +3824,7 @@ impl<'test> TestCx<'test> {
             .arg(format!("-Ldependency={}", &support_lib_deps_deps.to_string_lossy()))
             .arg("--extern")
             .arg(format!("run_make_support={}", &support_lib_path.to_string_lossy()))
+            .arg("--edition=2021")
```
Comment (jieyouxu) [inline src/tools/compiletest/src/runtest.rs:3827]: Hmm, do we want to force edition 2021 for *every* rmake test?

**Review by jieyouxu — APPROVED** [review]
The changes look good to me in general, only thing is I'm not 100% sure if we want to make *every* rmake.rs be edition 2021 by default. I think we can experiment with edition 2021 by default, and revert / check that individual rmake.rs can override with `//@ edition: 2015` if for some reason an earlier edition is required.

**Comment by jieyouxu** [comment]
Thanks for the PR!

@bors r+

File: src/tools/compiletest/src/runtest.rs (hunk 2 above)
Comment (fmease) [reply src/tools/compiletest/src/runtest.rs:3827]: I don't see why not. The `rmake` test files are “developer-facing” after all unlike the Rust files found in e.g. `tests/ui`. They are test runners, not “test subjects”. Lol, I'm lacking words. Similarly, `compiler/` / `rustc_*` crates all default to Rust 2021.

Comment (jieyouxu) [reply src/tools/compiletest/src/runtest.rs:3827]: Make sense yeah, I was just wondering if where was any weird edge cases I have not considered

**Comment by fmease** [comment]
Failed in rollup: https://github.com/rust-lang/rust/pull/124316#issuecomment-2074104431
@bors r-

**Comment by jieyouxu** [comment]
Thanks!

@bors r+ rollup

**Pull request merged**